src/parser/jobs/ast/modules/CelestialOpposition.tsx

```diff
@@ -47,40 +55,44 @@ export default class CelestialOpposition extends Module {
+			this.activeSect = this.data.actions.NOCTURNAL_SECT.id
+		}
 	}
 
 	onComplete() {
-		const holdDuration = this.uses === 0 ? this.parser.currentDuration : this.totalHeld
-		const usesMissed = Math.floor(holdDuration / this.data.actions.CELESTIAL_OPPOSITION.cooldown)
-		const fightDuration = this.parser.fight.end_time - this.parser.fight.start_time
-		const maxUses = (fightDuration / this.data.actions.CELESTIAL_OPPOSITION.cooldown) - 1
+		const holdDuration = this.uses === 0 ? this.parser.pull.duration : this.totalHeld
+		const missedUses = Math.floor(holdDuration / (this.data.actions.CELESTIAL_OPPOSITION.cooldown))
+		// TODO: update max uses to not use the whole duration such as when there are no available targets
+		const maxUses = (this.parser.pull.duration / this.data.actions.CELESTIAL_OPPOSITION.cooldown) - 1
 
 		const WASTED_USE_TIERS = {
-			[maxUses * SEVERITY_MOD.MINOR]: SEVERITY.MINOR,
-			[maxUses * SEVERITY_MOD.MEDIUM]: SEVERITY.MEDIUM,
-			[maxUses * SEVERITY_MOD.MAJOR]: SEVERITY.MAJOR, // if not used at all, it'll be set to 100 for severity checking
+			[maxUses * severityMod.minor]: SEVERITY.MINOR,
+			[maxUses * severityMod.medium]: SEVERITY.MEDIUM,
+			[maxUses * severityMod.major]: SEVERITY.MAJOR, // if not used at all, it'll be set to 100 for severity checking
 		}
 		const suggestContentDiurnal = <Trans id="ast.celestial-opposition.suggestion.content.diurnal">
-				Use <ActionLink {...this.data.actions.CELESTIAL_OPPOSITION} /> more frequently. In <StatusLink {...this.data.statuses.DIURNAL_SECT} />, the heal and regen combined add up to the same potency of a <ActionLink {...this.data.actions.BENEFIC_II} /> on each player it reaches.
+				Use <DataLink action="CELESTIAL_OPPOSITION" /> more frequently. In <DataLink status="DIURNAL_SECT" />, the heal and regen combined add up to the same potency of a <DataLink action="BENEFIC_II" /> on each player it reaches.
 				Trusting the regens to top off the party HP will save MP and GCDs on healing.
 		</Trans>
 		const suggestContentNoct = <Trans id="ast.celestial-opposition.suggestion.content.noct">
-				Use <ActionLink {...this.data.actions.CELESTIAL_OPPOSITION} /> more frequently. In <StatusLink {...this.data.statuses.NOCTURNAL_SECT} />, the shield is the same potency as from <ActionLink {...this.data.actions.ASPECTED_HELIOS} />,
+				Use <DataLink action="CELESTIAL_OPPOSITION" /> more frequently. In <DataLink status="NOCTURNAL_SECT" />, the shield is the same potency as from <DataLink action="ASPECTED_HELIOS" />,
 				so it can save MP and GCDs casting it. Since shields last 30 seconds it can be cast much earlier than incoming damage and allow the cooldown to refresh sooner.
 		</Trans>
 
-		const content = this.activeSect && this.activeSect.id === this.data.statuses.NOCTURNAL_SECT.id ? suggestContentNoct : suggestContentDiurnal
+		const content = this.activeSect !== -1 && this.activeSect === this.data.actions.NOCTURNAL_SECT.id ? suggestContentNoct : suggestContentDiurnal
```

Comment:
Mixing and matching actions with statuses. Still not sure on that -1 also.

Reply:
updated to be statuses